docs/content/_index.md

```diff
@@ -1,608 +1,1007 @@
+the expressions, the result is undefined.
+
+```live:example/iter/undefined:query:merge_down
+some i; input.servers[i].protocols[i] == "ssh"  # there is no assignment of i that satisfies the expression
+```
+```live:example/iter/undefined:output:expect_undefined
+```
+
+### Rules
+
+Rego lets you encapsulate and reuse logic with rules. Rules are just if-then
+logic statements. Rules can either be complete or incremental.
+
+#### Complete Rules
+
+Complete rules are if-then statements that assign a single value to a variable.
+For example:
+
+```live:example/complete/1:module:openable
+package example.rules
+
+any_public_networks = true {
+    net := input.networks[_]
+    net.public
 }
 ```
 
-As a result, any document, base or virtual, can be accessed hierarchically starting from the root data node – either as an identifier:
+Every rule consists of a _head_ and a _body_. In Rego we say the rule head is
+true _if_ the rule body is true for some set of variable assignments. In the
+example above `any_public_networks = true` is the head and `net :=
+input.networks[_]; net.public` is the body.
 
-```ruby
-import data.servers                            # Base document
-import data.opa.examples.violations            # Virtual document
+You can query for the value generated by rules just like any other value:
+
+```live:example/complete/1:query
+any_public_networks
+```
+```live:example/complete/1:output
 ```
 
-or as a URI component in an HTTP request:
+All values generated by rules can be queried via the global `data` variable.
 
-```http
-GET https://example.com/v1/data/servers HTTP/1.1
+```live:example/complete/1/abs:query:merge_down
+data.example.rules.any_public_networks
+```
+```live:example/complete/1/abs:output:merge_down
 ```
 
-```http
-GET https://example.com/v1/data/opa/examples/violations HTTP/1.1
+> 💡 You can query the value of any rule loaded into OPA by referring to it with an
+> absolute path. The path of a rule is always:
+> `data.<package-path>.<rule-name>`.
+
+If you omit the `= <value>` part of the rule head the value defaults to `true`.
+You could rewrite the example above as follows without changing the meaning:
+
+```live:example/complete_elided:module:read_only,openable
+package example.rules
+
+any_public_networks {
+    net := input.networks[_]
+    net.public
+}
 ```
 
-> Since the `data` document includes both base and virtual documents,
-> it is possible to query for both at the same time. The easiest way
-> to illustrate this is to query for _all_ of `data` at once. Note,
-> OPA does NOT allow base and virtual documents to overlap. For
-> example, if you try to load a rule that defines a virtual document
-> at path a/b/c (which is already defined by a base document), OPA
-> will return an error. Similarly, if you try to load a base document
-> into a path that is already defined by a virtual document, OPA will
-> also return an error.
+Similarly, if you omit the rule body it defaults to `true`. This allows you to
```

Comment:
Maybe clarify (re-iterate) what it means for the rule body to default to true. eg. this example it means that the statement in the head is always true (defined?), so `pi` will always be `3.14`.

Reply:
Done.